src/main/java/com/facebook/openwifirrm/modules/ApiServer.java

```diff
@@ -269,16 +290,23 @@ private String getFullUrl(String path, String queryString) {
 	 * HTTP 403 response and return false.
 	 */
 	private boolean performOpenWifiAuth(Request request, Response response) {
-		// TODO check if request came from internal endpoint
-		boolean internal = true;
-		String internalName = request.headers("X-INTERNAL-NAME");
-		if (internal && internalName != null) {
-			// Internal request, validate "X-API-KEY"
-			String apiKey = request.headers("X-API-KEY");
-			if (apiKey != null && apiKey.equals(serviceKey)) {
-				// auth success
-				return true;
+		boolean internal = request.port() == params.internalHttpPort;
+		if (internal) {
+			// TODO if this doesn't exist, is it an error???
```

Comment:
I think this IS an error and we'd just return unauthorized

Reply:
Yeah sure, though I don't understand the significance of this field...

Reply:
seems like all internal communications are expected to have this error so i'll leave it as an error and remove comment